src/perf/PerfListener.cc

```diff
@@ -92,25 +191,35 @@ void PerfListener::onPostForkParent(pid_t childPid) {
         attrs.sample_period = instructionCountLimit_ / samplingFactor_;
         attrs.wakeup_events = 1;
     }
-    // Apparently older (3.13) kernel versions doesn't support
-    // PERF_FLAG_FD_CLOEXEC. This fd will be closed anyway (by FilesListener) so
-    // it isn't very bad to not use it on newer kernels (and add it with fcnlt)
-    // until we implement some linux version discovery.
-    perfFd_ = withErrnoCheck(
-            "perf event open",
-            perf_event_open,
-            &attrs,
-            childPid,
-            -1,
-            -1,
-            PERF_FLAG_FD_NO_GROUP /* | PERF_FLAG_FD_CLOEXEC */);
-    withErrnoCheck("set cloexec flag on perfFd", fcntl, F_SETFD, FD_CLOEXEC);
-    if (instructionCountLimit_ != 0) {
-        int myPid = getpid();
-        withErrnoCheck("fcntl", fcntl, perfFd_, F_SETOWN, myPid);
-        int oldFlags = withErrnoCheck("fcntl", fcntl, perfFd_, F_GETFL, 0);
-        ;
-        withErrnoCheck("fcntl", fcntl, perfFd_, F_SETFL, oldFlags | O_ASYNC);
+
+    for (auto& config: eventConfigs) {
+        logger::debug("Opening perf event for pmu with id ", config.type);
+        attrs.type = config.type;
+        attrs.config = config.config[0];
+        attrs.config1 = config.config[1];
+        attrs.config2 = config.config[2];
+        // Apparently older (3.13) kernel versions doesn't support
+        // PERF_FLAG_FD_CLOEXEC. This fd will be closed anyway (by
+        // FilesListener) so it isn't very bad to not use it on newer kernels
+        // (and add it with fcnlt) until we implement some linux version
+        // discovery.
+        int perfFd = withErrnoCheck(
+                "perf event open",
+                perf_event_open,
+                &attrs,
+                childPid,
+                -1,
+                -1,
+                PERF_FLAG_FD_NO_GROUP /* | PERF_FLAG_FD_CLOEXEC */);
+        withErrnoCheck(
+                "set cloexec flag on perfFd", fcntl, F_SETFD, FD_CLOEXEC);
```

Comment:
```suggestion
        withErrnoCheck("set cloexec flag on perfFd", fcntl, F_SETFD, FD_CLOEXEC);
```
I prefer this like it was previosly

Reply:
Imho we shouldn't argue with clang-format.